src/Components/Switcher/Switcher.scss

```diff
@@ -12,68 +11,118 @@
     margin: 0;
   }
 
-  .toggle-switch {
-    appearance: none;
-    outline: none;
+  span {
     position: relative;
     cursor: pointer;
-    width: 22px;
-    height: 12px;
-    border-radius: 100px;
-    background: transparent;
+    margin-left: 30px;
+
+    &::before {
+      content: '';
+      position: absolute;
+      left: -30px;
+      top: 4px;
+      width: 22px;
+      height: 12px;
+      border-radius: 100px;
+      background: transparent;
+    }
 
     &::after {
       content: '';
       position: absolute;
-      top: 3px;
-      left: 3px;
-      bottom: 3px;
+      top: 7px;
+      left: -27px;
       width: 6px;
+      height: 6px;
       border-radius: 50%;
       transition: transform .25s ease;
     }
-  
-    &:checked::after {
-      transform: translateX(calc(22px - 13px + 1px));
-    }
-      
-    &+label {
-      cursor: pointer;
+  }
+
+  input[type="checkbox"] {
+    position: absolute;
+    opacity: 0;
+    visibility: hidden;
+    &:checked + span {
+      &::after {
+        transform: translateX(calc(22px - 13px + 1px));
```

Comment:
Are you sure, that we need to use calc function in this case?

Reply:
Yea. The span for the label has margin-left. The margin-left is for the decoration switcher. Because of this margin we need the transform for the pseudo-elements